main/ertp/api/contract-hosts.md

```diff
@@ -0,0 +1,153 @@
+```
+
+## contractHost.getInstallationSourceCode(installation)
+- `installation` `{Installation}`
+- Returns: `{String}`
+
+Get the actual source code of the Installation.
+
+```js
+const theSourceCode = E(someHost).getInstallationSourceCode(someInstallation);
+```
+
+## contractHost.redeem(invite)
+- `invite` `{Payment}`
+- Returns: `{Object}`
+
+Seat invitations are packaged as payments from the Invite Assay. Redeeming an invite returns a seat object with an arbitrary interface (the interface is at the discretion of the contract) which supports interaction according to the terms.
+
+```js
+const someSeat = E(someHost).redeem(someInvite);
+```
+
+## contractHost.getInviteAssay()
+- Returns: `{Assay}`
+
+The assay allows holders of seat invitations to get exclusive access to a Seat.
+
+```js
+const assayWithExclusiveAccess = E(someHost).getInviteAssay();
+```
+
+## installation.spawn(terms)
+- `terms` `{Terms}`
+- Returns: `{Invites <Object>}`
+
+Create a new `InviteMaker`, then call the Contract's `start()` method and return its results. The results are often a collection of seat invitations for the seats in the contract, but see [`coveredCall`](https://github.com/Agoric/ERTP/blob/master/core/coveredCall.js) for an exception.
+
+```js
+import harden from '@agoric/harden';
+
+const terms = harden({ left: whatPartyAWants, right: whatPartyBWants });
+const invitePayments = E(someInstallation).spawn(terms);
+```
+
+## installation.checkUnits(installation, inviteUnits, terms)
+- `installation` `{Installation}`
+- `inviteUnits` `{Units}`
+- `terms` `{Terms}`
+
+The writer of the contract can provide methods to help users of the contract verify that the terms of the contract match their expectation. These methods are defined with the installation as the first parameter, so the verifiers can validate that the caller's invitation was issued by the same one. The invocation by clients should omit this parameter, as they will be supplied with a copy of the function with that information already supplied.
+
+Users usually want to validate their invitation, the terms of the deal they're attempting to participate in, and which seat they are taking. If the invitation is invalid `checkUnits()` will throw an error.
+
+```js
+// If `allegedInviteUnits` is valid and matches the terms of the
+// contract, then the code will continue as expected.
+// Otherwise, `checkUnits()` will throw an error.
+E(escrowExchangeInstallationP)
+  .checkUnits(allegedInviteUnits, { left, right }, 'left')
+  .then(() => {
+    return E(inviteAssayP).claimExactly(
+      allegedInviteUnits,
+      allegedInvitePaymentP,
+      'verified invite',
+    );
+  });
+  ```
+
+## contract.start(terms, inviteMaker)
+- `terms` `{Terms}`
+- `inviteMaker` `{InviteMaker}`
+- Returns: `{Object}`
+
+Start execution of the contract. May return invites for seats.
+
+### <span style="color:red">Need help writing/checking example.</span>
```

Comment:
@katelynsills Can you check this example?

Reply:
adding to list for later